src/NHibernate/Engine/JoinSequence.cs

```diff
@@ -195,10 +195,17 @@ public JoinFragment ToJoinFragment(IDictionary<string, IFilter> enabledFilters,
 				{
 					// NH Different behavior : NH1179 and NH1293
 					// Apply filters for entity joins and Many-To-One association
-					var enabledForManyToOne = FilterHelper.GetEnabledForManyToOne(enabledFilters);
-					condition = new SqlString(string.IsNullOrEmpty(on) && (ForceFilter || enabledForManyToOne.Count > 0)
-					            	? join.Joinable.FilterFragment(join.Alias, enabledForManyToOne)
-					            	: on);
+					if (string.IsNullOrEmpty(on))
+					{
+						var enabledFiltersForJoin = ForceFilter ? enabledFilters : FilterHelper.GetEnabledForManyToOne(enabledFilters);
+						condition = new SqlString(ForceFilter || enabledFiltersForJoin.Count > 0
```

Comment:
I have considered `ForceFilter` was meant to apply all filters, not just many-to-one enabled filters. But maybe I am wrong and it is here just for discriminator filters and other non session filters related filtering. In such case this change may not be suitable.

`ForceFilter` is enabled for entity joins. I think entity joins are arbitrary joins and should not respect the many-to-one filter setting, which seems to me about entity association joins. In such case, this fix would be legit.

But if instead we consider the many-to-one filter setting should be taken into account for arbitrary joins, meaning the filters should be disabled on arbitrary joins if that setting is disabled, then we should instead just add a breaking change in the releases note for v5.3.0.